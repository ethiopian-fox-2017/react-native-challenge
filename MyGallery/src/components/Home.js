import React from 'react';
import Icon from 'react-native-vector-icons/Ionicons';
import { Header, Button, Body, Title, Text } from 'native-base';


export class Home extends React.Component {
  render() {
    return (
      <Header style={{backgroundColor: 'teal'}}>
        <Body>
          <Button>
            <Text>MyGallery</Text>
          </Button>
        </Body>
      </Header>
    );
  }
}

export default Home;
