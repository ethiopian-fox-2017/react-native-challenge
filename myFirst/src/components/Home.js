import React, { Component } from 'react';

import { Container, Content, } from 'native-base';
import { Provider } from 'react-redux';

import store from '../store/';
import styles from '../assets/index.js';
import Headernih from './Header';
import Contentnih from './Content';


export default class Home extends Component {

  render() {
    return (
      <Provider store={store}>
        <Container>
          <Headernih navigation={this.props.navigation}/>
          <Content style={styles.content}>
            <Contentnih />
          </Content>
        </Container>
      </Provider>
    );
  }
}
