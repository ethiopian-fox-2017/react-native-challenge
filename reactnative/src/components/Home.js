import React, { Component } from 'react';
import {
  StyleSheet,
  Text,
  View,
  Button
} from 'react-native';
import { StackNavigator } from 'react-navigation';

import Header from './Header';
import Content from './Content';
import Footer from './Footer';

class Home extends Component {
  constructor(props) {
    super(props)
    this.state = {
      name: 'Coldplay Albums'
    }
  }
  static navigationOptions = {
    title: 'Home',
    headerTitleStyle: {
       color: 'black',
       flex: 1,
       alignSelf: 'center',
       marginTop: 10
    },
    headerStyle: {
      backgroundColor: '#dddddd',
      flexDirection: 'row',
      justifyContent: 'center',
    }
  };
  render() {
    return (
      <View style={styles.container}>
        <Header nama={this.state.name} navigation={this.props.navigation}/>
        <Footer />
      </View>
    );
  }
}

const styles = StyleSheet.create({
  container: {
    flex: 2,
    flexDirection: 'column',
    alignItems: 'center',
    backgroundColor: '#fff'
  }
});

export default Home;
