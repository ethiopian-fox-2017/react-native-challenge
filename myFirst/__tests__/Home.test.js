import 'react-native';
import React from 'react';
import Home from '../src/components/Home';
import { shallow } from 'enzyme';
// Note: test renderer must be required after react-native.


test('renders correctly', () => {
  const tree = shallow(<Home />);
  expect(tree).toMatchSnapshot();
});
