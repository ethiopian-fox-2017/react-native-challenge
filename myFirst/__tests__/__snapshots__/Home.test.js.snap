// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <Provider
    store={
        Object {
            "dispatch": [Function],
            "getState": [Function],
            "replaceReducer": [Function],
            "subscribe": [Function],
            Symbol(Symbol.observable): [Function],
          }
    }
>
    <Styled(Container)
        virtual={undefined}
    >
        <Headernih
            navigation={undefined}
        />
        <Styled(Content)
            style={
                Object {
                    "paddingBottom": 20,
                  }
            }
            virtual={undefined}
        >
            <Connect(Content) />
        </Styled(Content)>
    </Styled(Container)>
</Provider>,
  "nodes": Array [
    <Provider
      store={
            Object {
                  "dispatch": [Function],
                  "getState": [Function],
                  "replaceReducer": [Function],
                  "subscribe": [Function],
                  Symbol(Symbol.observable): [Function],
                }
      }
>
      <Styled(Container)
            virtual={undefined}
      >
            <Headernih
                  navigation={undefined}
            />
            <Styled(Content)
                  style={
                        Object {
                              "paddingBottom": 20,
                            }
                  }
                  virtual={undefined}
            >
                  <Connect(Content) />
            </Styled(Content)>
      </Styled(Container)>
</Provider>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 0,
      "_context": Object {},
      "_currentElement": <Home />,
      "_debugID": 1,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": Home {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "props": Object {},
        "refs": Object {},
        "state": null,
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 1,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <Provider
          store={
                    Object {
                              "dispatch": [Function],
                              "getState": [Function],
                              "replaceReducer": [Function],
                              "subscribe": [Function],
                              Symbol(Symbol.observable): [Function],
                            }
          }
>
          <Styled(Container)
                    virtual={undefined}
          >
                    <Headernih
                              navigation={undefined}
                    />
                    <Styled(Content)
                              style={
                                        Object {
                                                  "paddingBottom": 20,
                                                }
                              }
                              virtual={undefined}
                    >
                              <Connect(Content) />
                    </Styled(Content)>
          </Styled(Container)>
</Provider>,
        "_debugID": 2,
        "_renderedOutput": <Provider
          store={
                    Object {
                              "dispatch": [Function],
                              "getState": [Function],
                              "replaceReducer": [Function],
                              "subscribe": [Function],
                              Symbol(Symbol.observable): [Function],
                            }
          }
>
          <Styled(Container)
                    virtual={undefined}
          >
                    <Headernih
                              navigation={undefined}
                    />
                    <Styled(Content)
                              style={
                                        Object {
                                                  "paddingBottom": 20,
                                                }
                              }
                              virtual={undefined}
                    >
                              <Connect(Content) />
                    </Styled(Content)>
          </Styled(Container)>
</Provider>,
      },
      "_renderedNodeType": 1,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <Home />,
}
`;
